### ISSUE TITLE
With Mockito 5 tests stall when Spring instantiates proxied configuration classes with mock in static field

### ISSUE THREAD

**Issue by hpoettker** [mockito/mockito#2999]
I've found a strange behavior when using Mockito 5 together with Spring. I'm not sure whether this is a bug in Mockito or in Spring but the problem does not occur with Mockito 4.

The following test fails due to the timeout as the instantiation of the proxied configuration class seems to stall:

```
@Test
@Timeout(value = 5, threadMode = SEPARATE_THREAD)
void stallsOnConfigClassInitialization() {
    new AnnotationConfigApplicationContext(ProxiedConfigurationClass.class);
}

@Configuration
static class ProxiedConfigurationClass {
    static Object object = Mockito.mock(Object.class);
}
```

If `proxyBeanMethods = false` is set on the configuration class, then the test won't stall, and thus succeed.

If the tests are run sequentially, first with proxying and then without proxying, then both tests will time out.

This behavior occurs with both Mockito 5.0.0 and 5.3.1 but not with 4.11.0.

Apart from Mockito, I'm using the following versions:

- OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)
- Spring Framework 6.0.8
- JUnit Jupiter 5.9.2
- Gradle 7.6.1

I've pushed a complete reproducer here: https://github.com/hpoettker/mockito-reproducer/blob/main/src/test/java/reproducer/MockitoReproducerTests.java

**Comment by TimvdLippe** [comment]
Please report this issue to the Spring folks.

**Comment by hpoettker** [comment]
I've reported the issue to in the Spring project: spring-projects/spring-framework#30438. It has been correctly identified as a duplicate of spring-projects/spring-framework#27180 and has been closed.

In turn, this makes this issue a duplicate of #2362.

That this problem appears only with Mockito 5 but not with Mockito 4 is explained well if the root cause is indeed the inline mock maker.

Would it make sense to open an issue in the byte buddy project?
